### ISSUE TITLE
Show tabs from only one container at the same time

### ISSUE THREAD

**Issue by bbenoit** [jonathanKingston/sea-containers#14]
This feature would achieve what some are used to do with the "_Tab Groups_" extension: see only the tabs grouped in a container at the same time.

Could be managed, for example, like layers in an image editor:
- A small eye icon near the container name (or the "+" symbol) to switch to the container's content with a click.
- Another click on the same icon brings back all tabs in the window.
- A click on the same icon for another container switches the window's content to the tabs in latter container only.

Anything close to that would be great!

**Comment by jonathanKingston** [comment]
So I assume you are referring to this extension: https://addons.mozilla.org/en-US/firefox/addon/tab-groups-panorama/

Could you screenshot the specific feature as it has a lot of functionality.

So I *think* you are after hiding all *personal* when in *work* for example? I was hoping to implement a global toggle for this feature. (This would behave similar to the Test Pilot's feature hide container) When a user changes to a Personal Tab, it would hide all other containers and show all personal tabs.

The performance of hiding and showing lots of tabs is pretty bad, currently there isn't a native API however lets add something and we can always try and make it native later 👍

**Comment by bbenoit** [comment]
> So I assume you are referring to this extension: https://addons.mozilla.org/en-US/firefox/addon/tab-groups-panorama/

Yes, indeed. But my idea is not to reimplement the whole concept, but just the fact that this add-on (_tab-groups-panorama_)allow to have a tab-bar with only tabs from a specific group at a time.

> Could you screenshot the specific feature as it has a lot of functionality.

Not really, as the suggested feature consist of just seeing only tabs relating to a group (or a container if transposed in the context of _sea-containers_) at once.

> So I think you are after hiding all personal when in work for example?

That is exactly what I tried to describe.

> This would behave similar to the Test Pilot's feature hide container

I wasn't aware of such a feature. Can't find any information about it (except an open feature request on GitHub).

> The performance of hiding and showing lots of tabs is pretty bad

I didn't see that. I naively thought of a (too ?) simple solution: if it is possible to apply a CSS rule to the tabs visible in the tab bar, "just" hide those that are not in the same container as the current one. But probably this solution is over-simplistic...

**Comment by jonathanKingston** [comment]
>  if it is possible to apply a CSS rule to the tabs visible in the tab bar

Currently the sidebar only shows one container at a time. To make the top bar match we actually need to hide the tabs. This requires a reload when they are reopened.

I'm currently trying to make keyboard shortcuts to the extension follow the tabs on the left rather than the normal order. Other than looking less cluttered is there any other need to hiding the top tabs?
(In this repos README I have a CSS change you can do to hide those tabs, although I will warn you, it will highlight more bugs in the extension hah).

As an extension myself I can't change the browser CSS though.

> I wasn't aware of such a feature
![selection_636](https://cloud.githubusercontent.com/assets/338988/26310654/881bfc2a-3ef9-11e7-83bb-b63f5b5cf931.png)

What you are asking for is kinda that but automatically when you enter and leave a container. For large amounts of pages it will likely be slow, it also has the risk of losing form data (currently).

Anyway, I keep asking for a platform API to hide these which would make implementing this without performance issues super simple.

**Comment by jonathanKingston** [comment]
Raised https://bugzilla.mozilla.org/show_bug.cgi?id=1366759 for implementing this within Firefox.

**Comment by CharoSW** [comment]
Would it be possible to have a semi solution where you list the containers in a drop-down menu (or similar) and only list the selected containers tabs in a box bellow?

**Comment by jonathanKingston** [comment]
@CharoSW interesting, so you are after only showing one container at all? I believe this would be a different solution entirely. The rationale to showing all container names is the quick overview of which window has what amount of containers and the easy access to creating new container tabs.
Changes within the container experiment may reduce the need for easy access to container tab creation so hiding the names would be probably helpful and even cleaner UX. (Can I create this as a separate issue?)

Current thinking is post 57 Firefox would allow extensions to disable the top tab strip which I think the current UX would remove the need for this bug entirely. Currently I auto hide the containers and besides the additional resources loaded there isn't an issue.
As mentioned I raised another bug about clearing up the resource issue / hiding the tabs themselves. This likely would be implemented after tab hiding anyway.

**Comment by CharoSW** [comment]
Mostly I am interested in a TabGroups replacement.  The Containers idea isn't exactly that, but it's a similar idea which with a bit of tweaking could probably deliver the same sort of effect.  You extension is a step in the direction I'd like things to go in because it shows one container and its tabs at a time without having to hide/unhide each container from the menu.  It thus allows me to get a rudimentary organisational system going where I can switch between all my tabs easily which is nice when you have over 60 tabs which relate to various work topics but also some downtime stuff like a webcomic I am in the middle of reading.   My suggestions was only to clean up the last of the post-tabgroup blues by showing only one container rather than the current list of them, with a way to switch between the containers like a drop-down box or maybe an initial menu that shows all the containers.  As such it's not so much the headings people wouldn't want, just the existence of another containers tabs in the same view.  I could probably get a mockup or similar of what I mean if I get a little more free time today.
